helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/AvroSchemaUtil.java

```diff
@@ -6,18 +6,75 @@
 
 package com.linkedin.avroutil1.compatibility;
 
-import java.lang.reflect.Field;
-import java.lang.reflect.Modifier;
-import java.util.IdentityHashMap;
-import java.util.List;
-import java.util.stream.Collectors;
 import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericContainer;
 import org.apache.avro.generic.GenericData;
 import org.apache.avro.specific.SpecificData;
+import org.apache.avro.specific.SpecificFixed;
 import org.apache.avro.specific.SpecificRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.IdentityHashMap;
+import java.util.List;
+import java.util.stream.Collectors;
 
 
 public class AvroSchemaUtil {
+  private final static Logger LOG = LoggerFactory.getLogger(AvroSchemaUtil.class);
+  private final static List<Schema.Type> INT_PROMOTIONS = Collections.unmodifiableList(Arrays.asList(
+          Schema.Type.LONG, Schema.Type.FLOAT, Schema.Type.DOUBLE
+  ));
+  private final static List<Schema.Type> LONG_PROMOTIONS = Collections.unmodifiableList(Arrays.asList(
+          Schema.Type.FLOAT, Schema.Type.DOUBLE
+  ));
+  private final static Schema NONE = Schema.create(Schema.Type.NULL); //marker
+  private final static ClassValue<Schema> DECLARED_SCHEMAS = new ClassValue<Schema>() {
+    @Override
+    protected Schema computeValue(Class<?> type) {
```

Comment:
Is there a more descriptive name we can use for this method? I am a bit confused, by name alone, as to what `computeValue` does